client/src/components/add-item/add-item.tsx

```diff
@@ -0,0 +1,130 @@
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useForm } from "react-hook-form";
+import { z } from "zod";
+
+import { apiClient } from "@/api-client";
+import { Button } from "@/components/ui/button";
+import {
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from "@/components/ui/dialog";
+import {
+  Form,
+  FormControl,
+  FormDescription,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "@/components/ui/form";
+import { Input } from "@/components/ui/input";
+import { useState } from "react";
+
+const formSchema = z.object({
+  mainText: z.string().min(1, {
+    message: "Main text must be at least 1 characters.",
+  }),
+  subText: z.string().optional(),
+});
+
+export default function AddItem() {
+  const [open, setOpen] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+  const form = useForm<z.infer<typeof formSchema>>({
+    resolver: zodResolver(formSchema),
+    defaultValues: {
+      mainText: "",
+      subText: "",
+    },
+  });
+
+  // 2. Define a submit handler.
```

Comment:
If this comment is not essential, I would recommend you to remove this comment :)